src/libraries/Common/src/System/Net/Http/aspnetcore/Quic/Implementations/Managed/Internal/RangeSet.cs

```diff
@@ -0,0 +1,336 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Net.Quic.Implementations.Managed.Internal.Frames;
+using System.Net.Security;
+using System.Numerics;
+
+namespace System.Net.Quic.Implementations.Managed.Internal
+{
+    /// <summary>
+    ///     Container for storing numbers with efficient range queries and insertions. The values are maintained in a list of ranges.
+    /// </summary>
+    internal class RangeSet : IEnumerable<RangeSet.Range>
+    {
+        private List<Range> _ranges;
```

Comment:
Sadly, I also need access to previous/next item, I am not aware of any collection that would allow me to do so.

the way it is used now, it is expected to contain exactly one element most of the time. More items should occur only when packets are lost or reordered, or this were deliberately targeted by malicious peer.